### ISSUE TITLE
Mailfush problem in 

### ISSUE THREAD

**Issue by filternetz** [globaleaks/GlobaLeaks#1497]
ii  globaleaks                           2.60.116                            Opensource whistleblowing platform.
DISTRIB_DESCRIPTION="Ubuntu 12.04.5 LTS"

When I updated to the latest globaleaks version. I keep having a problem with sending out email and getting this error in globaleaks.log.

2015-11-24 14:53:18+0100 [-] &quot;[!] Exception while performing scheduled operation MailflushSchedule: &#39;progressive&#39;&quot;
2015-11-24 14:53:18+0100 [-] &#39;[!] Unhandled exception raised:&#39;
2015-11-24 14:53:18+0100 [-] &#39;[!] KeyError Mapping key not found.&#92;&#92;n&#92;&#92;nTraceback (most recent call last):&#92;&#92;n&#92;&#92;n  File &quot;&#47;usr&#47;lib&#47;python2.7&#47;dist-packages&#47;globaleaks&#47;jobs&#47;base.py&quot;, line 102, in _operation&#92;&#92;n    yield self.operation()&#92;&#92;n&#92;&#92;nKeyError: &#92;&#39;progressive&#92;&#39;&#92;&#92;n&#39;
2015-11-24 14:53:18+0100 [-] Starting factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x43f8908&gt;
2015-11-24 14:53:21+0100 [ParserProtocol,client] Stopping factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x43f8908&gt;

Hope this helps.

**Comment by filternetz** [comment]
Some more debug logs.

2015-11-24 14:58:08+0100 [-] &#39;[!] Warning: [Mailflush] is taking more than 5 minutes to execute&#39;
2015-11-24 14:58:08+0100 [-] Starting factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x422cfc8&gt;
2015-11-24 14:58:11+0100 [ParserProtocol,client] Stopping factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x422cfc8&gt;
2015-11-24 14:58:17+0100 [-] &quot;[!] Exception while performing scheduled operation MailflushSchedule: &#39;progressive&#39;&quot;
2015-11-24 14:58:17+0100 [-] &#39;[!] Unhandled exception raised:&#39;
2015-11-24 14:58:17+0100 [-] &#39;[!] KeyError Mapping key not found.&#92;&#92;n&#92;&#92;nTraceback (most recent call last):&#92;&#92;n&#92;&#92;n  File &quot;&#47;usr&#47;lib&#47;python2.7&#47;dist-packages&#47;globaleaks&#47;jobs&#47;base.py&quot;, line 102, in _operation&#92;&#92;n    yield self.operation()&#92;&#92;n&#92;&#92;nKeyError: &#92;&#39;progressive&#92;&#39;&#92;&#92;n&#39;
2015-11-24 14:58:17+0100 [-] Starting factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x88fad88&gt;
2015-11-24 14:58:20+0100 [ParserProtocol,client] Stopping factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x88fad88&gt;

**Comment by IuliiSe** [comment]
Same here. We also experience the issues with outgoing mails in v116

**Comment by evilaliv3** [comment]
gonna work on this! having a community using on it is helping with more testing the new releases :)

i will update you both on this ticket by few hours having pushed a bugfixed release.

**Comment by evilaliv3** [comment]
this has been fixed and shippeed in 2.60.117

**Comment by filternetz** [comment]
I upgraded to 2.60.117 and the issue hasn't been solved for me. See attached log:

```
2015-11-30 13:30:42+0100 [-] Received SIGTERM, shutting down.
2015-11-30 13:30:42+0100 [-] (TCP Port 8082 Closed)
2015-11-30 13:30:42+0100 [-] Stopping factory &lt;cyclone.web.Application instance at 0x3d54518&gt;
2015-11-30 13:30:42+0100 [-] Main loop terminated.
2015-11-30 13:30:42+0100 [-] Server Shut Down.
2015-11-30 13:30:42+0100 [-] Exiting GlobaLeaks
2015-11-30 13:35:13+0100 [-] Starting GlobaLeaks
2015-11-30 13:35:13+0100 [-] Loading &#47;usr&#47;lib&#47;python2.7&#47;dist-packages&#47;globaleaks&#47;backend.py...
2015-11-30 13:35:13+0100 [-] Loaded.
2015-11-30 13:35:13+0100 [-] Log opened.
2015-11-30 13:35:13+0100 [-] twistd 13.2.0 (&#47;usr&#47;bin&#47;python 2.7.3) starting up.
2015-11-30 13:35:13+0100 [-] reactor class: twisted.internet.epollreactor.EPollReactor.
2015-11-30 13:35:13+0100 [-] Application starting on 8082
2015-11-30 13:35:13+0100 [-] Starting factory &lt;cyclone.web.Application instance at 0x3a671b8&gt;
2015-11-30 13:35:13+0100 [-] switching group privileges since 0 to 1337
2015-11-30 13:35:13+0100 [-] switching user privileges since 0 to 1337
2015-11-30 13:35:13+0100 [-] Found an already initialized database version: 24
2015-11-30 13:35:15+0100 [-] [+] Hardwriting hidden service in the DB: http:&#47;&#47;5karyquenden4d6k.onion
2015-11-30 13:35:15+0100 [-] [!!] Public Website (https:&#47;&#47;secure.publeaks.nl) is not automatically overwritten by (https:&#47;&#47;5karyquenden4d6k.tor2web.org)
2015-11-30 13:35:15+0100 [-] [ ] GLBackend is now running
2015-11-30 13:35:15+0100 [-] [ ] Visit http:&#47;&#47;127.0.0.1:8082 to interact with me
2015-11-30 13:35:15+0100 [-] [ ] Visit http:&#47;&#47;localhost:8082 to interact with me
2015-11-30 13:35:15+0100 [-] [ ] Visit http:&#47;&#47;5karyquenden4d6k.onion to interact with me
2015-11-30 13:35:15+0100 [-] [ ] Visit https:&#47;&#47;secure.publeaks.nl to interact with me
2015-11-30 13:35:15+0100 [-] [ ] Remind: GlobaLeaks is not accessible from other URLs, this is strictly enforced
2015-11-30 13:35:15+0100 [-] [ ] Check documentation in https:&#47;&#47;github.com&#47;globaleaks&#47;GlobaLeaks&#47;wiki&#47; for special enhancement
2015-11-30 13:35:25+0100 [-] &#39;[!] Exception while performing scheduled operation MailflushSchedule: %d format: a number is required, not str&#39;
2015-11-30 13:35:25+0100 [-] &#39;[!] Unhandled exception raised:&#39;
2015-11-30 13:35:28+0100 [-] &#39;[!] TypeError Inappropriate argument type.&#92;&#92;n&#92;&#92;nTraceback (most recent call last):&#92;&#92;n&#92;&#92;n  File &quot;&#47;usr&#47;lib&#47;python2.7&#47;dist-packages&#47;globaleaks&#47;jobs&#47;base.py&quot;, line 104, in _operation&#92;&#92;n    yield self.operation()&#92;&#92;n&#92;&#92;nTypeError: %d format: a number is required, not str&#92;&#92;n&#39;
2015-11-30 13:35:28+0100 [-] Starting factory &lt;txsocksx.client.SOCKS5ClientFactory instance at 0x3857200&gt;
```

**Comment by evilaliv3** [comment]
than

**Comment by evilaliv3** [comment]
issue fixed on Publeaks setup; general refactor applied to current devel branch would totally address what was causing the bug in previous releases migrations and will be shipped in 2.60.115.

**Comment by marceloomens** [comment]
We've again run into this problem in the Netherlands, so I guess it's now safe to say that it's not due to an upgrade-related issue. 

As before this will probably work:

    TRUNCATE TABLE eventlogs

but that's not exactly a long-term solution. Apparently every so often a notification is generated that blocks the mailqueue, for whatever reason, leading to no notifications being delivered by the node at all. I find it hard to believe that we're the only ones with this problem.

BTW is this related to https://github.com/globaleaks/GlobaLeaks/issues/1532 or not?

**Comment by evilaliv3** [comment]
exactly that was!

by the way the fix is right only for versions < 2.60.127

current version reimplent completely the mail flush sched and does not make use of any archived json data so it's resilient to this issue by design!
